icgcget/cli.py

```diff
@@ -71,6 +55,38 @@ def logger_setup(logfile):
     return logger
 
 
+def check_code(client, code):
+    if code != 0:
+        logger.error("{} client exited with a nonzero error code {}.".format(client, code))
+        raise click.ClickException("Please check client output for error messages")
```

Comment:
This may be silly, but click exceptions are used to handle expected errors because they exit cleanly.  They're all raised from the main method because I didn't think importing them all over the codebase was very clean.